src/test/regress/sql/shard_rebalancer.sql

```diff
@@ -1462,6 +1462,42 @@ DROP VIEW table_placements_per_node;
 DELETE FROM pg_catalog.pg_dist_rebalance_strategy WHERE name='capacity_high_worker_2';
 DELETE FROM pg_catalog.pg_dist_rebalance_strategy WHERE name='only_worker_1';
 
+-- add colocation groups with shard group count < worker count
+-- the rebalancer should balance those "unbalanced shards" evenly as much as possible
+SELECT 1 FROM citus_remove_node('localhost', :worker_2_port);
+create table table_with_one_shard_1 (a int primary key);
+create table table_with_one_shard_2 (a int primary key);
+create table table_with_one_shard_3 (a int primary key);
```

Comment:
Since we're colocating these three tables together I think it's good to name them with a different pattern than the 4 and 5 tables below. e.g.

```
single_shard_colocation_1a
single_shard_colocation_1b
single_shard_colocation_1c
single_shard_colocation_2a
single_shard_colocation_2c
```